js/src/forum/components/Composer.js

```diff
@@ -64,12 +66,10 @@ export default class Composer extends Component {
 
       this.prevPosition = this.state.position;
     }
+  }
 
-    if (isInitialized) return;
-
-    // Since this component is a part of the global UI that persists between
-    // routes, we will flag the DOM to be retained across route changes.
-    context.retain = true;
+  oncreate(vnode) {
```

Comment:
Why? It currently animates in just fine... Besides, since this.prevPosition is set to this.state.position oninit, it wouldn't be applicable?

Reply:
I see how the `if` branch won't be executed in `oncreate` (as `this.prevPosition` will probably be undefined), but what about the `else` branch? Without this, `this.prevPosition` won't be set on creation.

Reply:
Well, `this.prevPosition` is set on `oninit`, and `animatePositionChange` isn't relevant, because the state's position is always hidden on creation (since this is the "composer manager", and not the composer body.

Reply:
Ah, I missed your point about `this.prevPosition` being initialized. More importantly, though, `oncreate` does not need any of the height animation stuff, because the composer is always hidden on first render.